File: apps/ios/ShadowChatCoreBridge/RedactedLogger.swift
import Foundation

struct RedactedLogger {
    func redactUsername(_ value: String) -> String {
        let trimmed = value.trimmingCharacters(in: .whitespacesAndNewlines)
        guard trimmed.count > 2 else { return "[redacted]" }
        return "\(trimmed.prefix(2))***"
    }

    func redactPhone(_ value: String?) -> String {
        guard let value, !value.isEmpty else { return "[none]" }
        return "***\(value.suffix(2))"
    }

    func redactToken(_ value: String) -> String {
        return "tok_***\(value.suffix(4))"
    }

    func info(_ message: String) {
        #if DEBUG
        NSLog("[ShadowChat][INFO] %@", message)
        #endif
    }

    func warning(_ message: String) {
        #if DEBUG
        NSLog("[ShadowChat][WARN] %@", message)
        #endif
    }
}

File: apps/android/app/src/main/java/com/shadowchat/RedactedLogger.kt
package com.shadowchat

import android.util.Log

object RedactedLogger {
    private const val TAG = "ShadowChat"

    fun redactUsername(value: String): String {
        val trimmed = value.trim()
        return if (trimmed.length > 2) "${trimmed.take(2)}***" else "[redacted]"
    }

    fun redactPhone(value: String): String {
        return "***${value.takeLast(2)}"
    }

    fun redactToken(value: String): String {
        return "tok_***${value.takeLast(4)}"
    }

    fun info(message: String) {
        Log.i(TAG, message)
    }

    fun warn(message: String) {
        Log.w(TAG, message)
    }
}
